core/src/main/scala/org/ensime/core/AnalyzerManager.scala

```diff
@@ -183,7 +133,23 @@ class AnalyzerManager(
 
     }
 
-  private def withExistingModuleFor(file: Path, req: RpcAnalyserRequest)(f: (RpcAnalyserRequest, EnsimeProject) => Unit): Unit = {
+  def toFile(f: SourceFileInfo) = f.file match {
+    case RawFile(path) => path.toFile
+    case ArchiveFile(path, _) => path.toFile
```

Comment:
I'm not sure if `.toFile` is correct for `ArchiveFile`. We may be missing a test case here. How come you want to get the `File` out?